---
title: "proteomics-tici-rat-astrocytes"
author: "pp"
date: "1/14/2022"
output: pdf_document
---
The goal of this project is to analyze the data from the proteomics study of TICI-treated rat astrocytes. There are 5 samples (biological replicates) corresponding to TICI-treated astrocytes and 5 samples that are Vehicle-treated astrocytes. A total of 1596 proteins were identified. VEH2 sample was excluded due to having low ion intensities (probably due to low cell count) making it a total of 9 samples including both TICI and Veh.

Calling all the packages
```{r}
library(dplyr)
library(tidyverse)
library(edgeR)
library(ggfortify)
library(plyr)
library(ggplot2)
library(scales)
library(grid)
library(pheatmap)
library(RColorBrewer)
library(ggeasy)
library(EnhancedVolcano)
library(ggrepel)
```

Calling the ggbiplot_script
```{r}
source("ggbiplot_script.R")
```

Loading data from the proteomics raw tmt output file.
```{r}
protein_groups <- read_csv("../data/tmt_astro_raw_data.csv")
protein_groups_tmt <- protein_groups %>%
  select(protein_ids = 'gene_names',
         TICI1 = 'TICI1',
         TICI2 = 'TICI2',
         TICI3 = 'TICI3',
         TICI4 = 'TICI4',
         TICI5 = 'TICI5',
         VEH1 = 'VEH1',
         VEH3 = 'VEH3',
         VEH4 = 'VEH4',
         VEH5 = 'VEH5'
  )

```

Check if the data follows a negative binomial distribution.
```{r}
intensity.plot <- protein_groups_tmt %>%
  ggplot() +
  geom_histogram(aes(TICI1, fill = "TICI", color = "1"), bins = 100) +
  geom_histogram(aes(TICI2, fill = "TICI", color = "2"), bins = 100) +
  geom_histogram(aes(TICI3, fill = "TICI", color = "3"), bins = 100) +
  geom_histogram(aes(TICI4, fill = "TICI", color = "4"), bins = 100) +
  geom_histogram(aes(TICI5, fill = "TICI", color = "5"), bins = 100) +
  geom_histogram(aes(VEH1, fill = "VEH", color = "1"), bins = 100) +
  geom_histogram(aes(VEH3, fill = "VEH", color = "3"), bins = 100) +
  geom_histogram(aes(VEH4, fill = "VEH", color = "4"), bins = 100) +
  geom_histogram(aes(VEH5, fill = "VEH", color = "5"), bins = 100) +
  scale_y_continuous(trans = "log1p", breaks = c(0, 1000, 2000, 4000, 6000)) +
  labs(title="Distribution of ion intensity",x="TMT intensity", y = "Count")
```

Creating the initial edgeR object with all the TMT values from each sample group. Here I will analyze the DE proteins in TICI vs. VEH samples.
```{r}
groups = c("TICI", "TICI", "TICI", "TICI", "TICI", "VEH", "VEH", "VEH", "VEH") %>%
  factor(levels = c("TICI", "VEH"))
design = model.matrix(~0+groups)
contrasts = makeContrasts(
  H = groupsTICI - groupsVEH,
  levels = design
)
TMT_edgeR <- DGEList(
  counts = protein_groups_tmt %>%
    mutate(protein_ids = make.unique(protein_ids)) %>%
    na.omit() %>%
    column_to_rownames("protein_ids"),
  group = c(1,1,1,1,1,2,2,2,2)
)
```

Filtering out rows so that each row has at least 9 non-zero values.
```{r}
TMT_edgeR_9 <- TMT_edgeR[rowSums(cpm(TMT_edgeR) == 0) <= (9-9) , ]
TMT_edgeR_9 <- TMT_edgeR_9 %>% calcNormFactors()
TMT_edgeR_9 <- TMT_edgeR_9 %>% estimateGLMCommonDisp(design=design)
lrt_TMT_9 <- glmLRT(glmFit(TMT_edgeR_9, design), contrast = contrasts) %>% decideTestsDGE(p.value = 0.05)
plotSmear(TMT_edgeR_9, de.tags = rownames(lrt_TMT_9)[lrt_TMT_9 %>% as.logical()], main = 'MA plot for count data')
abline(h=c(-1,1), col='blue')

#new analysis added
de <- exactTest(TMT_edgeR_9)
de.proteins <- row.names(topTags(de, n=500)$table)
plotSmear(TMT_edgeR_9, de.tags=de.proteins, main='Top 500 proteins')
```

Making the PCA plot
```{r}
make_pca_plot <- function(tp, tp_edger, ellipse = TRUE) {
  tp %>%
    na.omit %>%
    filter(protein_ids %in% tp_edger) %>%
    column_to_rownames("protein_ids") %>%
    as.matrix() %>%
    t %>%
    prcomp(center = T, scale = T) %>%
    ggbiplot(labels = c("TICI1", "TICI2", "TICI3", "TICI4", "TICI5", "VEH1", "VEH3", "VEH4", "VEH5"),
             groups = c("TICI", "TICI", "TICI", "TICI", "TICI", "VEH", "VEH", "VEH", "VEH"),
             ellipse = ellipse)+
    theme_classic()
  }
```

View the PCA plot
```{r}
pca.plot <- protein_groups_tmt %>%
  make_pca_plot(tp_edger = rownames(lrt_TMT_9)[as.logical(lrt_TMT_9)]) +
  ggtitle("TICI astrocytes")
ggeasy::easy_center_title()
```

The PC1 and PC2 separate very well and explain the variation in ~85% of the data. Now I will create the tables.
```{r}
table_TMT_edgeR <- (glmLRT(glmFit(TMT_edgeR_9, design), contrast = contrasts) %>% 
                      topTags(n=2000))$table %>%
  rownames_to_column("gene_names") %>%
  as_tibble() %>%
  merge(protein_groups %>%
          mutate(`gene_names` = 
                   make.unique(`gene_names`))) %>%
  as_tibble() %>%
  arrange(FDR)

write.table(table_TMT_edgeR, file = "../data/results.csv", sep = ",", row.names = FALSE)
```

Making the heatmap - Wrong code
```{r}
# df <- read.csv('../data/results.csv')
# df <- df[1:63, ]
# #df[ , 7:15][df[, 7:15] == 0] <- 1
# 
# df %>% mutate(proteins = make.unique(gene_names)) %>%
#   filter(FDR<0.1) %>%
#   select(-logFC,	-logCPM,	-LR,	-PValue,	-FDR,	-peptide_no,	-percent_sequence_coverage,	-mol_weight,	-q_value,	-score,	-intensity,	-msms_count,	-protein_names, -protein_ids) %>%
#   mutate(mean1 = mean(c(TICI1, TICI2, TICI3, TICI4, TICI5, VEH1, VEH3, VEH4, VEH5))) %>%
#   mutate_if(is.numeric, log10) %>%
#   mutate_if(is.numeric, list(~ . - mean1)) %>%
#   mutate(sd1 = sd(c(TICI1, TICI2, TICI3, TICI4, TICI5, VEH1, VEH3, VEH4, VEH5))) %>%
#   mutate_if(is.numeric, list(~ . / sd1)) %>%
#   select(-gene_names, -mean1, -sd1) ->
#   vals
# 
# plot <- vals %>%
#   column_to_rownames("proteins") %>%
#   as.matrix() %>%
#   pheatmap::pheatmap(color = colorRampPalette(rev(brewer.pal(n = 7, name = "RdBu")))(100),
#                      main = "TICI Astrocyte Top Proteins",
#                      cluster_rows = TRUE,
#                      cluster_cols = TRUE, legend = TRUE, breaks = seq(-0.1, 0.1, length.out=100), legend_breaks = NA, legend_labels = NA, annotation = NA, annotation_colors = NA, annotation_legend = TRUE, fontsize = 10, fontsize_row = 10)

```

Making heatmap by calculating Z-scores with the scale function
```{r}
df <- read.csv('../data/results.csv')
#df <- df[1:63, ]

log10.intensity.df = df %>% mutate(proteins = make.unique(gene_names)) %>%
  filter(FDR<0.1) %>%
  select(-logFC,	-logCPM,	-LR,	-PValue,	-FDR,	-peptide_no,	-percent_sequence_coverage,	-mol_weight,	-q_value,	-score,	-intensity,	-msms_count,	-protein_names, -protein_ids) %>%
  mutate(mean1 = mean(c(TICI1, TICI2, TICI3, TICI4, TICI5, VEH1, VEH3, VEH4, VEH5))) %>%
  mutate_if(is.numeric, log10)

log10.intensity.df[,2:10] = t(scale(t(log10.intensity.df[,2:10])))

rownames(log10.intensity.df) = log10.intensity.df$proteins

log10.intensity.df = log10.intensity.df[,2:10]


heat.map <- log10.intensity.df %>% as.matrix() %>%
  pheatmap::pheatmap(color = colorRampPalette(rev(brewer.pal(n = 7, name = "RdBu")))(100),
                     main = "TICI Astrocyte DE Proteins",
                     cluster_rows = TRUE,
                     cluster_cols = TRUE, breaks = seq(-2, 2, length.out=100), legend = TRUE, legend_breaks = NA, legend_labels = NA, annotation = NA, annotation_colors = NA, annotation_legend = TRUE, fontsize = 10, fontsize_row = 10)
```

Making volcano plot
```{r}
p <- ggplot(data=table_TMT_edgeR, aes(x=logFC, y=-log10(PValue), col=gene_names, label=gene_names)) + 
  geom_point() + 
  theme_minimal() +
geom_text_repel() +
        geom_vline(xintercept=c(-0.6, 0.6), col="red") +
        geom_hline(yintercept=-log10(0.05), col="red")

mycolors <- c("blue", "red", "black")
names(mycolors) <- c("DOWN", "UP", "NO")
p3 <- p + scale_colour_manual(values = mycolors)

p3      
```

Using biomaRT 
```{r}
library(biomaRt)

# function for converting human gene symbol to mouse gene symbol; retrieves one-to-one, one-to-many, and many-to-one orthologs
human <- useEnsembl(biomart = "ensembl", 
                    dataset = "hsapiens_gene_ensembl", 
                    mirror = "useast")
mouse <- useEnsembl(biomart = "ensembl", 
                    dataset = "mmusculus_gene_ensembl", 
                    mirror = "uswest")

convertHumanGeneList <- function(x){
  require(biomaRt)
  genesV2 = getLDS(attributes = c("hgnc_symbol"), filters = "hgnc_symbol", values = x , mart = human, attributesL = c("mgi_symbol"), martL = mouse, uniqueRows=T, )
  return(genesV2)
}

convert.df <- convertHumanGeneList(VECTOR_OF_GENES_TO_CONVERT) # replace with a vector of your genes you wish to convert
colnames(convert.df) <- c("gene", "mus_gene")
convert.df <- convert.df[!(duplicated(convert.df$gene) | duplicated(convert.df$gene, fromLast = TRUE)), ] # remove all many to one orthologs
convert.df <- convert.df[!(duplicated(convert.df$mus_gene) | duplicated(convert.df$mus_gene, fromLast = TRUE)), ] # remove all one to many orthlogs
converted.genes <- convert.df$mus_gene # get the new mouse gene names
converted.genes <- converted.genes[!is.na(converted.genes)] # make sure no NAs snuck through
```

